src/js/_enqueues/lib/delayed-inline-script-loader.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * Handles delayed execution of before/after inline scripts for defer/async scripts.
+ *
+ * @output wp-includes/js/wp-delayed-inline-script-loader.js
+ */
+
+(function (window, document) {
+	var nonce = document.currentScript.nonce,
+		doneDependencies = new Set();
+
+	/**
+	 * Determines whether a script was loaded.
+	 *
+	 * @param {string} dep Dependency handle.
+	 * @returns {boolean} Whether dependency was done.
+	 */
+	function isDependencyDone(dep) {
+		return doneDependencies.has(dep);
+	}
+
+	/**
+	 * Runs an inline script.
+	 *
+	 * @param {HTMLScriptElement} script Script to run.
+	 */
+	function runInlineScript(script) {
+		var newScript;
+		script.dataset.wpDone = "1";
+		if (nonce && nonce !== script.nonce) {
+			console.error(
+				"CSP nonce check failed for after inline script. Execution aborted.",
+				script
+			);
+			return;
+		}
+		newScript = script.cloneNode(true);
+		newScript.type = "text/javascript";
+		script.parentNode.replaceChild(newScript, script);
+	}
+
+	/**
+	 * Runs the supplied inline scripts if all of their dependencies have been done.
+	 *
+	 * @param {NodeList<HTMLScriptElement>} scripts Scripts to run if ready.
+	 */
+	function runReadyInlineScripts(scripts) {
+		var i, len, deps;
+		for (i = 0, len = scripts.length; i < len; i++) {
+			deps = scripts[i].dataset.wpDeps.split(/,/);
+			if (deps.every(isDependencyDone)) {
+				runInlineScript(scripts[i]);
+			}
+		}
+	}
+
+	/**
+	 * Runs whenever a load event happens.
+	 *
+	 * @param {Event} event Event.
+	 */
+	function onScriptLoad(event) {
+		var matches, handle, script;
+		if (
+			!(
+				event.target instanceof HTMLScriptElement ||
+				event.target.async ||
+				event.target.defer ||
+				event.target.id
+			)
+		) {
+			return;
+		}
+
+		// TODO: Consider adding a data attribute which specifically contains the handle.
```

Comment:
@westonruter I'm glad you raised the idea of the data attribute to contain the handle, as this was an initial thought pattern I had way back in discovery, that I had forgotten about. This would also negate the duplicate ID issue, not so?

Reply:
I've captured this in #70. So we can remove this now:

```suggestion
```